src/material-experimental/mdc-slider/slider.ts

```diff
@@ -22,13 +75,229 @@ import {SliderAdapter} from './slider-adapter';
   selector: 'mat-slider',
   templateUrl: 'slider.html',
   styleUrls: ['slider.css'],
+  host: {
+    'class': 'mat-mdc-slider mdc-slider',
+    '[class.mdc-slider--range]': 'isRange',
+    '[class.mdc-slider--disabled]': 'isDisabled',
+    '[class.mdc-slider--discrete]': 'isDiscrete',
+    '[class.mdc-slider--tick-marks]': 'hasTickMarks',
+  },
+  exportAs: 'matSlider',
   changeDetection: ChangeDetectionStrategy.OnPush,
   encapsulation: ViewEncapsulation.None,
 })
-export class MatSlider {
+export class MatSlider implements AfterViewInit, OnDestroy {
+  /** The slider thumb(s). */
+  @ViewChildren('thumb') _thumbs: QueryList<ElementRef<HTMLElement>>;
 
-  /** Instance of the MDC slider foundation for this slider. */
+  /** The slider thumb knob(s) */
+  @ViewChildren('knob') _knobs: QueryList<ElementRef<HTMLElement>>;
+
+  /** The active section of the slider track. */
+  @ViewChild('trackActive') _trackActive: ElementRef<HTMLElement>;
+
+  /** The sliders hidden range input(s). */
+  @ContentChildren(MatSliderThumb, { descendants: false }) _inputs: QueryList<MatSliderThumb>;
+
+  /** Whether the slider is disabled. */
+  @Input()
+  get isDisabled(): boolean { return this._isDisabled; }
+  set isDisabled(v: boolean) {
+    this._isDisabled = coerceBooleanProperty(v);
+    if (this._initialized) {
+      this._foundation.setDisabled(v);
+    }
+  }
+  private _isDisabled: boolean = false;
+
+  /** Whether the slider displays a numeric value label upon pressing the thumb. */
+  @Input()
+  get isDiscrete(): boolean { return this._isDiscrete; }
+  set isDiscrete(v: boolean) { this._isDiscrete = coerceBooleanProperty(v); }
+  private _isDiscrete: boolean = false;
+
+  /** Whether the slider displays tick marks along the slider track. */
+  @Input()
+  get hasTickMarks(): boolean { return this._hasTickMarks; }
```

Comment:
What was the rationale for changing this API from the current `MatSlider` (which uses `tickInterval` for this? I also think the API would probably be something like `showTickMarks`

Reply:
I didn't implement `tickInterval` because it is not part of the MDC implementation. We'd have to add that functionality ourselves

I'm ok with renaming this to `showTickMarks` (again, I was just copying the variable name from [the MDC foundation](https://github.com/material-components/material-components-web/blob/master/packages/mdc-slider/foundation.ts#L65))